src/internal/proc.js

```diff
@@ -481,7 +485,17 @@ export default function proc(
 
     try {
       suspend()
-      const task = proc(taskIterator, subscribe, dispatch, getState, options, effectId, fn.name, (detached ? null : noop))
+      const task = proc(
+        taskIterator,
+        subscribe,
+        dispatch,
+        getState,
+        (detached ? null : taskContext), // TODO: topContext should be passed in case of spawn?
```

Comment:
yeah but you could also view `spawn` as a simple child task with different behavior regarding cancellation/errors? And also with analogy to lexical scope, it gets its params from its parent task, so could be the same from dynamic scope (context)